### PR TITLE
Expose AccessControlConfig in TestingTrinoServer bindings

### PR DESCRIPTION
**Pull request by bitweld**
Add AccessControlConfig binding to TestingTrinoServer to reflect product runtime environment and allow other consumers to access it.

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/testing/TestingAccessControlManager.java [hunk 1]
```diff
@@ -125,9 +125,14 @@
     private BiPredicate<Identity, String> denyIdentityTable = IDENTITY_TABLE_TRUE;
 
     @Inject
+    public TestingAccessControlManager(TransactionManager transactionManager, EventListenerManager eventListenerManager, AccessControlConfig accessControlConfig)
+    {
+        super(transactionManager, eventListenerManager, accessControlConfig, DefaultSystemAccessControl.NAME);
+    }
+
     public TestingAccessControlManager(TransactionManager transactionManager, EventListenerManager eventListenerManager)
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/testing/TestingAccessControlManager.java:133]: why do you need this constructor?

Comment (ksobolew) [inline core/trino-main/src/main/java/io/trino/testing/TestingAccessControlManager.java:133]: `@VisibleForTesting`?

**Review by ksobolew — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/testing/TestingAccessControlManager.java (hunk 1 above)
Comment (bitweld) [reply core/trino-main/src/main/java/io/trino/testing/TestingAccessControlManager.java:133]: There are ~6 tests where this class is instantiated directly (w/o guice injection), I've decided to leave this "old" constructor to not pollute existing code with new argument that is not required nor have special usage.

Comment (bitweld) [reply core/trino-main/src/main/java/io/trino/testing/TestingAccessControlManager.java:135]: These classes are in general for testing only so this annotation could be misleading IMHO

**Review by ksobolew — APPROVED** [review]

**Pull request merged**